Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -637,12 +678,153 @@ namespace xr
                 }
             }
 
-            // Provide suggested bindings to instance
-            XrInteractionProfileSuggestedBinding suggestedBindings{ XR_TYPE_INTERACTION_PROFILE_SUGGESTED_BINDING };
-            XrCheck(xrStringToPath(instance, ActionResources.DEFAULT_XR_INTERACTION_PROFILE, &suggestedBindings.interactionProfile));
-            suggestedBindings.suggestedBindings = bindings.data();
-            suggestedBindings.countSuggestedBindings = (uint32_t)bindings.size();
-            XrCheck(xrSuggestInteractionProfileBindings(instance, &suggestedBindings));
+
+            // Create controller get trigger value action and suggested bindings
+            {
+                XrActionCreateInfo actionInfo{ XR_TYPE_ACTION_CREATE_INFO };
+                actionInfo.actionType = XR_ACTION_TYPE_FLOAT_INPUT;
+                strcpy_s(actionInfo.actionName, ActionResources.CONTROLLER_GET_TRIGGER_VALUE_ACTION_NAME);
+                strcpy_s(actionInfo.localizedActionName, ActionResources.CONTROLLER_GET_TRIGGER_VALUE_ACTION_LOCALIZED_NAME);
+                actionInfo.countSubactionPaths = static_cast<uint32_t>(ActionResources.ControllerSubactionPaths.size());
+                actionInfo.subactionPaths = ActionResources.ControllerSubactionPaths.data();
+                XrCheck(xrCreateAction(ActionResources.ActionSet, &actionInfo, &ActionResources.ControllerGetTriggerValueAction));
+                // For each controller subaction
+                for (size_t idx = 0; idx < ActionResources.CONTROLLER_SUBACTION_PATH_PREFIXES.size(); ++idx)
+                {
+                    // Create suggested binding
+                    std::string path{ ActionResources.CONTROLLER_SUBACTION_PATH_PREFIXES[idx] };
+                    path.append(ActionResources.CONTROLLER_GET_TRIGGER_VALUE_PATH_SUFFIX);
+                    bindings.push_back({ ActionResources.ControllerGetTriggerValueAction });
+                    XrCheck(xrStringToPath(instance, path.data(), &bindings.back().binding));
```

Comment:
What happens if a controller input profile doesn't support this path? I'm assuming the `XrCheck` throws and we're left in a bad state?

Reply:
i thought all the Windows MR controllers have a trigger button. Also, one more question i have is - do we want to support other MR controllers? because right now, we are only supporting windows MR controllers

Reply:
I think it's okay for us to only currently add in the WMR controllers, but since this is the OpenXR module it should be able to eventually support non-WMR controllers as well.